src/format.ts

```diff
@@ -115,101 +116,132 @@ function checkFormat(srcFiles: string[], baseArgs: string[], options: Options):
                           }
                         },
                         ['ignore', 'pipe', process.stderr])
-                    .stdout; 
+                    .stdout;
     out.setEncoding('utf8');
     out.on('data', (data: Buffer) => {
-      output += data; 
+      output += data;
     });
-    
-    out.on('end', () => {
-      findFormatErrorLines(output, options)
-        .then(() => {
-          resolve(output.indexOf('<replacement ') === -1 ? true : false);
-        });
+
+    out.on('end', async () => {
+      const files = output.split('<?xml version=\'1.0\'?>\n');
+      for (let i = 1; i < files.length; i++) {
+        const replacements = getReplacements(files[i]);
+        if (replacements.length > 0) {
+          const diff = await getDiffObj(srcFiles[i - 1], replacements)
+          printDiffs(diff, options);
+        }
+      }
+      resolve(output.indexOf('<replacement ') === -1 ? true : false);
     });
   });
 }
 
 /**
- * Parses through xml string for the replacement offsets and lengths. Uses those values
- * to locate the formatting error lines.
- * 
- * @param output xml string 
- * @param options
+ * Parses through xml string for the replacement string, offset, length.
+ *
+ * @param output xml string from clangFormat
  */
-async function findFormatErrorLines(output: string, options: Options){
-  const files = output.split('<?xml version=\'1.0\'?>\n');
-
-  for(let i = 1; i < files.length; i++){
+function getReplacements(output: string): replacement[] {
+  const replacements: replacement[] = [];
+  const offsets: string[]|null = output.match(/(?<=offset=\')(\d+)(?=\')/g);
```

Comment:
I think the approach in this function would benefit from an inline comment.

Reply:
done. Check to see if it's ok